doc/sphinx/practical-tools/utilities.rst

```diff
@@ -56,22 +56,30 @@ options to ``coqc`` (e.g. ``-R``, ``-Q``, ``-I``, see :ref:`command
 line options <command-line-options>`). Collecting the list of files
 and options is the job of the ``_CoqProject`` file.
 
+A ``_CoqProject`` file may contain the following kinds of entries in any order,
+separated by whitespace:
+* Selected options of coqc, which are forwarded to it directly. Currently these
+are ``-Q``, ``-I``, ``-R`` and ``-native-compiler``.
+* ``-arg`` options for other options of coqc that don’t fall in the above set.
+* Options specific to coq_makefile. Currently this is ``-docroot``.
+* Paths to files belonging to the project.
+* Comments, started by an unquoted ``#`` and going on until the end of the
+line.
+
 A simple example of a ``_CoqProject`` file follows:
 
 ::
 
     -R theories/ MyCode
-    -arg -w
-    -arg all
+    -arg "-w all"
```

Comment:
May I suggest to keep
```
-arg -w
-arg all
```
or transform it to
```
-arg -w -arg all
```
because this workaround is known to work well with build systems and IDEs.
See e.g. https://github.com/coq/coq/issues/5773 and related PRs & issues.
Perhaps this should be documented in the manual.

Please ignore my comment if this has been recently fixed in at least `coq_makefile`, CoqIDE, and ProofGeneral.

Reply:
In this example I tried to declare that "best practice" in a _CoqProject file might use double-quotes instead of unquoted arguments.

For coq_makefile and ProofGeneral, `-arg -w -arg all` and `-arg "-w all"` are equivalent (before and after this PR). The issue that CoqIDE had is probably fixed in this PR, but I haven’t tested it.

Reply:
This should indeed be tested.

Reply:
Tested it and could verify that CoqIDE works. What I did boils down to:
```bash
mkdir -p /tmp/A/theories
cat > /tmp/A/theories/A.v << EOF
Test Default Goal Selector.
EOF
cat > /tmp/A/_CoqProject << EOF
-R theories/ T
-arg "-w default"
-arg "-set 'Default Goal Selector=!'"
theories/A.v
EOF
```
Then open `/tmp/A/theories/A.v` in CoqIDE and step through the file.